From 2994307fe092a9627e12ad7cd9f32f4d36c201d8 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sun, 25 Sep 2022 15:58:37 +0200
Subject: [PATCH] mt76: mt7915: fix changed IEEE80211_MAX_AMPDU_BUF define in
 new kernel

New kernel use IEEE80211_MAX_AMPDU_BUF_HE instead of
IEEE80211_MAX_AMPDU_BUF.
This got backported to stable kernel 5.15.61 and cause compilation
error.
Add required ifdef to handle this changed define.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 mt7915/init.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mt7915/init.c b/mt7915/init.c
index 489f6e77..1813370a 100644
--- a/mt7915/init.c
+++ b/mt7915/init.c
@@ -327,8 +327,8 @@ mt7915_init_wiphy(struct ieee80211_hw *hw)
 	struct mt7915_dev *dev = phy->dev;
 
 	hw->queues = 4;
-	hw->max_rx_aggregation_subframes = IEEE80211_MAX_AMPDU_BUF_HE;
-	hw->max_tx_aggregation_subframes = IEEE80211_MAX_AMPDU_BUF_HE;
+	hw->max_rx_aggregation_subframes = IEEE80211_MAX_AMPDU_BUF;
+	hw->max_tx_aggregation_subframes = IEEE80211_MAX_AMPDU_BUF;
 	hw->netdev_features = NETIF_F_RXCSUM;
 
 	hw->radiotap_timestamp.units_pos =
-- 
2.37.2
